src/main/java/org/lantern/Proxifier.java

```diff
@@ -138,30 +138,11 @@ public synchronized void onSystemProxyChanged(
      * @param pce The proxy connection event.
      */
     @Subscribe
-    public synchronized void onProxyConnectionEvent(
-        final ProxyConnectionEvent pce) {
+    public synchronized void onProxyConnectedEvent(
+            final ProxyConnectedEvent pce) {
         this.lastProxyConnectionEvent = pce;
-        final ConnectivityStatus stat = pce.getConnectivityStatus();
-        switch (stat) {
-        case CONNECTED:
-            LOG.debug("Got connected event");
-            proxyWithChecks();
-            break;
-        case CONNECTING:
-            LOG.debug("Got connecting event");
-            break;
-        case DISCONNECTED:
-            LOG.debug("Got disconnected event");
-            try {
-                stopProxying();
-            } catch (final ProxyConfigurationError e) {
-                LOG.warn("Could not unproxy?", e);
-            }
-            break;
-        default:
-            break;
-        
-        }
+        LOG.debug("Got connected event");
+        proxyWithChecks();
```

Comment:
Again, once the proxy is connected, it's connected.  We never learn about disconnects anymore.